algorithms/closest_pair.py

```diff
@@ -84,7 +84,12 @@ def _distance_between(point_1, point_2):
 
```

Comment:
Re: the double loop in `_calculate_closest_split_pair`.

For your point given by the `i` index, you want to look at most at the next 7 points (that's the weird/clever part of the algorithm - the closest points MUST be no more than 7 index's away). When you get towards the end of the list of `potential_closest_split_pairs` though, an index which is 7 greater than `i` may well be out or range. So by setting `j` to the minimum of 7 or the length of `potential_split_pairs` - `i`, you can guarantee that index  `i` + `j` will always be in range.
And yes it is confusing! Maybe I should find a way to neaten it up.